verification/cbmc/proofs/aws_byte_cursor_advance/aws_byte_cursor_advance_harness.c

```diff
@@ -36,7 +36,9 @@ void aws_byte_cursor_advance_harness() {
     } else {
         assert(rv.ptr == old.ptr);
         assert(rv.len == len);
-        assert(cursor.ptr == old.ptr + len);
+        if (old.ptr) {
+            assert(cursor.ptr == old.ptr + len);
+        }
```

Comment:
Same for all similar cases. Can we say the following?
```c
assert(AWS_IMPLIES(old.ptr != NULL, cursor.ptr == old.ptr + len));
```

Reply:
Hmm, it seems to me like we are losing something here after this change.
Now we don't seem to check anything if `old_cur.ptr` is `NULL`.

How about:
```c
if (old_cur.ptr) {
    assert(cursor.ptr == old_cur.ptr + len);
} else {
    assert(cursor.ptr == NULL);
}
```